Docker and Docker Compose update for Windows and new documents for Linux (#607)
OctopusDeploy/docs: docs/installation/octopus-in-container/octopus-server-container-linux.md

---
title: Octopus Deploy Server Container Linux
description: An Octopus Deploy Server instance can be run directly from within a container.
position: 1
---

> If you're looking to run an Octopus Deploy Windows container, please refer to the [Octopus Deploy Server Container Windows](/docs/installation/octopus-in-container/octopus-server-container-windows.md) documentation.

Reviewer: Slight tweak:
```suggestion
If you want to run an Octopus Deploy Windows container, please refer to the [Octopus Deploy Server Container Windows](/docs/installation/octopus-in-container/octopus-server-container-windows.md) documentation.
```

> Running the Octopus Deploy Server inside a container provides a simple way to set up an Octopus Deploy instance. Upgrading to the latest version of Octopus is just a matter of running a new container with the new image version.

Reviewer: ```suggestion
Running the Octopus Deploy Server inside a container provides a simple way to set up an Octopus Deploy instance, and upgrading to the latest version of Octopus is just a matter of running a new container with the new image version.
```

Setting `--name OctopusServer` gives us an easy to remember name for this container. This is optional, but we recommend you provide a name that is meaningful to you, as that will make it easier to perform actions on the container later if necessary.

Using `--publish 1322:8080` maps the _container port_ `8080` to `1322` on the host so that the Octopus instance is accessible outside this sever.

To set the connection string we provide an _environment variable_ `DB_CONNECTION_STRING` (this can be to a local database or an external database).

In this example, we are running the image `octopusdeploy/octopusdeploy:2019.13.4`. The tag maps directly to the Octopus Server version that is bundled inside the image.

## Configuration

When running an Octopus Server Image, the following values can be provided to configure the running Octopus Server instance.

### Environment Variables

Read the Docker [docs](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) about setting environment variables.

|  Name       |    |
| ------------- | ------- |
|**DB_CONNECTION_STRING**|Connection string to the database to use|

> |**MASTER_KEY**|The master key to use to connect to an existing database. If not supplied, and the database does not exist, it will generate a new one. If the database does exist, this is mandatory|

Reviewer: ```suggestion
|**MASTER_KEY**|The master key to use to connect to an existing database. If not supplied, and the database does not exist, it will generate a new one. The master key is mandatory if the database exists|
```

> |**OCTOPUS_SERVER_BASE64_LICENSE**|Your License key for Octopus Deploy. If left empty, it will try and create a free license key for use

Reviewer: ```suggestion
|**OCTOPUS_SERVER_BASE64_LICENSE**|Your license key for Octopus Deploy. If left empty, it will try and create a free license key for use
```

|**ADMIN_USERNAME**|The admin user to create for the Octopus Server|
|**ADMIN_PASSWORD**|The password for the admin user for the Octopus Server|
|**ADMIN_EMAIL**|The email associated with the admin user account|

### Exposed Container Ports
Read Docker [docs](https://docs.docker.com/engine/reference/commandline/run/#publish-or-expose-port--p---expose) about exposing ports.

|  Port       |    |
| ------------- | ------- |
|**8080**| Port for API and HTTP portal |
|**10943**|Port for Polling Tentacles to contact the server|

> _The Octopus Server container does not currently support HTTPS however this should be available sometime in the future_

Reviewer: ```suggestion
_The Octopus Server container does not currently support HTTPS, however, this should be available sometime in the future_
```
